### PR TITLE
Rename B64 to Base64, Rfc2045 to Base64_rfc2045

### PR DESCRIPTION
**Pull request by dinosaure**

### CODE REVIEW

File: src/dune [hunk 1]
```diff
@@ -1,13 +1,13 @@
 (library
  (name base64)
- (modules b64)
+ (modules base64)
  (wrapped false)
```
Comment (avsm) [inline src/dune]: can remove `(wrapped false)` now

**Comment by avsm** [comment]
The README also still has a reference to B64

**Comment by timbertson** [comment]
Just want to make sure I'm not missing something: back in 2014, https://github.com/mirage/ocaml-base64/pull/4 renamed `Base64` -> `B64` in the first place to avoid naming clashes with other libraries. Now that's effectively been reverted, but there's no solution for the naming clashes that happen at link time (other than "don't use extlib or core_extended.base64")?

I saw https://github.com/lpw25/namespaces via https://discuss.ocaml.org/t/what-to-do-about-files-both-define-module-base64/3426/2, but it sounds like this is an eventual solution, so in the meantime users of cohttp + extlib need to pin this libeary to >1.x but <3.x so thy get a version of this library when it was named B64?

**Comment by dinosaure** [comment]
@timbertson I understand your concerns and it's look like a regression if you consider #4. The initial problem is a misspelling name between B64 (the module) and Base64 (the package) which cames from a trick between ocamlfind (META file) and OPAM (opam file) - where it's not needed to share the same name between the module provided and the package provided.

I think, for newcomers to the OCaml community, it's a barrier where the question: where is the Base64 module? becomes a repeating problem (see #17).

Currently, we have no solution about that and the problem mostly concerns `extlib` than `core_extended.base64` (where you can just remove this dependency and use others `core_extended.*`).

About `extlib`, I can try to figure out a way (which will break compatibility) to allow to use it in same time than `ocaml-base64`. However, from what I know (feedbacks from some packages releasers- but not all), `extlib` is less used - and this break let to developpers to reconsider utility of `extlib` inside their projects (like [`infer`](https://github.com/facebook/infer/pull/1056)).

So, if you really need `extlib`, I can figure out about a proper solution between these packages 😃 you just need to tell me.

**Comment by timbertson** [comment]
Yeah, that's fair reasoning. It was mostly just super confusing when I was sure that I had a version new enough to be after fix for this clash, and then found out it was also a version new enough to reintroduce it.

I do need both extlib and cohttp (for https://github.com/timbertson/passe, it's not on opam). The frustrating thing is I just got done ripping out batteries (I started this project a while ago when I was new to OCaml, and batteries friction with parts of the stdlib wasn't worth it in retrospect). I ended up keeping extlib just for its `OptParse` module (which batteries copied wholesale, so it's compatible). I don't actually need the rest of extlib, I'd happily take a standalone version of optparse. I'll see if the extlib authors are open to that...

**Pull request merged**